### ISSUE TITLE
PRIV_TOP / PRIV_TASK mixed use broken in 6.3

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#3505]
This bug was found in the context of very weird PRIV_TOP behavior, which was finally boiled down to a testcase where a `PRIV_TOP` was guaranteed to exist, and yet could not be retrieved later. To cut a long story short, I will jump right to the conclusion:

[vrt_priv_dyncmp()](https://github.com/varnishcache/varnish-cache/blob/6.3/bin/varnishd/cache/cache_vrt_priv.c#L99-L102) does not provide total ordering because `vmod_id` and `id` are treated equally. The fix is obvious and will be provided.

This ticket is to provide relevant information about the bug:

* It is only relevant if different `id` values are used, which, for all practical purposes, means that it will only surface when `PRIV_TOP` and `PRIV_TASK` are both used.
* The bug surfaces in existing PRIVs not being found during lookup (and suplicate PRIVs for the same id being created).

History: The bug was introduced with d6d3416090b8b7db62d2cd0958e756e23ea92352 in an effort to fix a regression which I introduced with 86af5ce095a6e39e313cf7c6c23a49edba4aace6 (which was actually even worse than this bug because `PRIV_TOP` and `PRIV_TASK` entries were confused). The bug was fixed for 6.4 / master with 6dfcc79d708d9013a07aab8f135003323b607bca as part of the actual fix to #3003

But notice that, unfortunately, 6.4 and 6.5 are also broken for a different reason, see #3496 (but this bug can be avoided by ensuring to access all PRIV_TOPs at esi_level 0, which can be considered a reliable workaround for all practical purposes).

So, in short, PRIV_TOP was broken in some way or the other between 25349b1ac78e8eed1e2da98ed941ad058232019b and 3228cb70159a77a677945f4520e9c7b51edb6e95
